blenderproc/python/types/MeshObjectUtility.py

```diff
@@ -337,6 +337,32 @@ def update_from_bmesh(self, bm: bmesh.types.BMesh, free_bm_mesh=True) -> bmesh.t
         # Make sure the mesh is updated
         self.get_mesh().update()
 
+    def join_with_other_objects(self, objects: List[Self]):
+        """
+            Joins the given list of objects with this object.
+
+            Does not change the global selection.
+
+        :param objects: List of objects which will be merged with this object
+        """
+        # save selection for undo selection in this function
+        selected_objects = bpy.context.selected_objects
+        # deselect all current selected objects
+        bpy.ops.object.select_all(action='DESELECT')
+        # set the current object as the joined target
+        self.select()
+        bpy.context.view_layer.objects.active = self.blender_obj
+        # select all objects which will be merged with the joined target
+        for obj in objects:
+            obj.select()
+        # execute the joined operation
+        bpy.ops.object.join()
+        bpy.ops.object.select_all(action='DESELECT')
+
+        # undo selection in this function
+        for obj in selected_objects:
+            MeshObject(obj).select()
```

Comment:
Instead of performing the tedious selection/deselection, it is also possible to choose the objects to join via the context dict given to the blender ops. Its sometimes a bit hard to find out which keys to set, but if one finds them, it makes the whole code easier, faster and less error prone.

```suggestion
        context = {}
        context["object"] = context["active_object"] = self.blender_obj
        context["selected_objects"] = context["selected_editable_objects"] = [obj.blender_obj for obj in objects] + [self.blender_obj]
        bpy.ops.object.join(context)
```